### PR TITLE
Character Attributes - Part one

### PR DESCRIPTION
**Pull request by MaxIsJoe**
image

In this rework I'm approaching things a little bit more differently, instead of replacing an entire core system from the game and force everyone to catch our bugs for us; this major rework is opt-in until we iron out all our bugs and slowly transition to this new approach we're taking.

The character attributes rework is meant to give us the ability to easily create forks that deviate from standard /tg/ job and role designs and give designers, server owners and admins more freedom and ease of use to create their own antagonists, roles and character traits, etc. with relative ease. This rework will allow us to add things like quirks into unitystation, give more powers to admemes and custom admin events, factions and perks for the upcoming Fallout 13 in unitystation project and a xeno evolution trees for UnityMarines.

For this first part of this rework, I've mainly added the ability for players to spawn in normally. But in future PRs, there will be more experimental togglable features while we slowly axe the old hardcoded systems and replace it with the new modular one.

### Changelog:

CL: [Improvement] Some icons have had their anti-alising disabled so they look a lot better now.
CL: [New] Added the base for character attributes, this rework is opt-in inside the job selection screen only currently until it's 100% done.

### CODE REVIEW

**Comment by Fr-Dae** [comment]
Job2
Is it possible to have something readable like this please?

**Comment by MaxIsJoe** [comment]
> Job2 Is it possible to have something readable like this please?

this is a backend change, it does not touch any UI stuff and I don't intend on doing so anytime soon.
maybe after character attributes are done i could look into it.

**Comment by Fr-Dae** [comment]
Thanks for the information, I'll see if anyone in my team has a prototype compatible with your work, I've heard of an xml grid to manage alignment and colour. I'll have to look into that. 
good luck with your draft =)

**Comment by MaxIsJoe** [comment]
> Thanks for the information, I'll see if anyone in my team has a prototype compatible with your work, I've heard of an xml grid to manage alignment and colour. I'll have to look into that. good luck with your draft =)

Occupation SO is going to be absolute and nuked sometime in the future, character attributes still doesn't have a behavior object ready for defining extra data on attributes yet.
Just keep that in mind while you work on this.

**Comment by Fr-Dae** [comment]
Thank you, I am passing on this information, it is of a technical level that is beyond me, the translation does not help.

**Comment by MaxIsJoe** [comment]
It's finally ready for review and for us to test drive it on staging

**Comment by MaxIsJoe** [comment]
image

I think tests are drunk

**Comment by MaxIsJoe** [comment]
image

yeah, they were. re-ran the tests and suddenly no errors.
can someone re-run tests on github?

**Comment by MaxIsJoe** [comment]
did someone re-run tests like i asked?

**Comment by corp-0** [comment]
I will now

**Comment by MaxIsJoe** [comment]
looks like tests are still failing for some reason, will open it soon to see what its still complaining about.

**Comment by MaxIsJoe** [comment]
image

I think this branch is cursed

**Comment by MaxIsJoe** [comment]
there's a different error every time i run tests and i can't find the issues it keeps reporting about

**Comment by MaxIsJoe** [comment]
image
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by MaxIsJoe** [comment]
finally

**Comment by MaxIsJoe** [comment]
done with codacy issues, the remaining issues is codacy being annoying

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 1]
```diff
@@ -210,8 +212,13 @@ public static GameObject ServerClonePlayer(Mind forMind, Vector3Int worldPositio
 	private static Vector3Int GetSpawnPointForOccupation(Occupation occupation)
 	{
 		Transform spawnTransform;
+		if (occupation == null)
+		{
+			spawnTransform = SpawnPoint.GetRandomPointForJob(JobType.ASSISTANT);
+			return spawnTransform != null ? spawnTransform.position.CutToInt() : Vector3Int.zero;
+		}
 		//Spawn normal location for special jobs or if less than 2 minutes passed
-		if (GameManager.Instance.stationTime < ARRIVALS_SPAWN_TIME || occupation.LateSpawnIsArrivals == false)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:214]: why remove the second condition?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:214]: Whoops, forgot to add this back in while testing. re-added it back.

File: UnityProject/Packages/packages-lock.json [hunk 2]
```diff
@@ -106,7 +106,7 @@
       "url": "https://packages.unity.com"
     },
     "com.unity.scriptablebuildpipeline": {
-      "version": "1.19.6",
+      "version": "1.20.1",
```
Comment (corp-0) [inline UnityProject/Packages/packages-lock.json]: is this an unintended change?

**Comment by corp-0** [comment]
assuming it works, I didn't see anything else worth mentioning.

File: UnityProject/Packages/packages-lock.json (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Packages/packages-lock.json]: this is intended.

Comment (MaxIsJoe) [reply UnityProject/Packages/packages-lock.json]: oh wait, i thought this was this #9291 PR.
well either way it gets updated in that PR as well so its fine

**Comment by MaxIsJoe** [comment]
hewwo?

**Comment by Bod9001** [comment]
we have to merge the unity update before this?

**Comment by MaxIsJoe** [comment]
> we have to merge the unity update before this?

we don't need to

**Review by corp-0 — APPROVED** [review]

**Pull request merged**